### ISSUE TITLE
hardcore mode: escape pods, permanent death, etc.

### ISSUE THREAD

**Issue by Nikwigi** [endless-sky/endless-sky#362]
I posted these in steam as well. but I thought here might be good too. 

Auto landing options
Back in evn some of the most fun was the stress of land while under fire. that sense of relief after making it. auto landing takes much of that threat away. maybe a simple toggle in settings for this would be nice.

Resale price settings
I'd like to see loss when you sell your ships and such. Once you get a nice ship it gets pointless fast capturing pirate ships and selling them back at full price. Also buying and selling your ships to get to this point has no sense of risk if you buy a ship at any point you simply sell as you move your way up the list. 
Might be part of the reason why Escape Velocity only let you own one ship.
I'd love to be able to pick a percentage like easy normal hard. Something set at the start would be best so you cant change it as you please. Maybe just a slider bar. from 10%-100%. This would add point to the more hardcore players. Maybe two sliders for Outfits and and Ships resale price. so the real hardcore players can even set ship resale to 0% if they so please.

Death permanent setting.
Adding the setting and escape pods would add point for those hardcore players.

**Comment by Thunderforge** [comment]
I would be in favor of auto-landing as a preference, with the default to on.

Difficulty settings would make sense, but I think a lot more discussion needs to happen about what that means. The most obvious way would be weighting damage in favor of either the player or the enemies. A more advanced way would be to implement a system of [slotting](http://gamedev.stackexchange.com/a/112041/52205) on easier difficulties, where the AI sometimes ignores the player or does something stupid in order to give the player a break.

As for death permanent as a setting, I always felt that was something that nobody actually used. Besides, someone being hardcore can just delete their save if they really want to do that. But if others would actually use it, I would not object to adding it in.

**Comment by Nikwigi** [comment]
Yeah I only knew a few that did I used it once in a while.  And while I agree with easy normal and hard for damage settings. I wouldn't want resale value tied in with it. I hate to have to play hard mode just to have my resale prices where I want them.

**Comment by endless-sky** [comment]
The auto-landing AI only takes control if you are not moving your ship, so anyone who wants to land manually can do so.

Having ships depreciate is a reasonable idea, but it's already part of a separate feature request. (#428)

So, I'm shifting this issue to focus on "hardcore mode." That will mainly involve adding:
* Escape pods
* Ideally, actually show your escape pod getting rescued and brought to a planet.
* Either they must always drop you on a planet with ships for sale, or there must also be a way to "book a flight" to another planet, as a passenger.

**Comment by Thunderforge** [comment]
I think that one thing that would be necessary for a "hardcode mode" would be that death is permanent if you don't have an escape pod. In Escape Velocity, it meant that loading the pilot file would get a message that the pilot is killed and there isn't anything that could be done to change that. With Endless Sky having save points, things get messy. We either:

 - Only "kill" the latest savepoint and all previous ones are still valid. This diminishes the lethality of hardcore mode because there is still a way to get back your character (and may encourage excessive creation of savepoints)
 - "Kill" every savepoint so that the character is not recoverable in any way, shape, or form. I think that there will be some who think this is unfair, especially if they wanted to put a savepoint before a major storyline that way they could reuse a hardcode mode character.

I also still wonder if people will actually use a hardcore mode. I don't recall it being that commonly used by EV players and I suspect that most Endless Sky players will be in the camp of "I don't mind it being there, but I won't use it." If that's the case, then I think our efforts would be better spent elsewhere.

**Comment by erprice** [comment]
Or disable save points for hardcore characters. This would make it much more rogue-like.

**Comment by erprice** [comment]
Another thing to consider is how much money the player has. If you are heavily in debt and your ship blows up, you might get dropped onto a nearby planet with a shipyard, but if you can't afford a new ship your game is over anyway.

Maybe the simplest way to do it is to say that your insurance paid out, then show the popup letting you select from the three starter ships. This will allow you to drop the player on the nearest inhabited planet, regardless of shipyards.

**Comment by Kastores** [comment]
+1 for this idea, and some suggestions.  I would definitely use hardcore mode.  For me, the game is much more exciting when there is genuine risk.  However, as a way to mitigate that somewhat, would it be possible to introduce an "insurance" mechanic to cover the loss of a ship up to a certain amount?  That way, provided you make it safely back in an escape pod, you're not totally out of cash.  I'm thinking something like the system EVE has in place.

Also, @endless-sky, it might be easier just to automatically dump a saved escape pod and character on the nearest planet with a shipyard.  I don't think you'd have to bother with "booking passage."  Just cover the journey with some flavor text that says, basically, "after several weeks of thumbing a ride, you arrive at Planet X."  As I recall (it's been a while) that's how Escape Velocity handled this situation.

Great game, so far.  Thanks for all the hard work!

**Comment by Tadrix** [comment]
Why isn't considered that escape pod could be picked up by other ship in the fleet, and it would make that ship a temporary flagship? Also, it shouldn't be the only pod — captain should make sure the crew has evacuated before saving himself.

**Comment by sivael** [comment]
I like the "escape pod" approach as that would mean you simply continue onwards. Would need to rebalance the cost of new ships vs players balance, maybe introduce some kind of insurance payout in case of ship destruction?

What I would not like is to force a player to one way of playing. Escape pods could make the whole game more streamlined and immersive, and they could as well make people quit. BUT. With the three last saves this rarely is an issue since if someone wants to respawn by loading, they can.

Thus, escape pod idea is great and would love to have it in game.
Problem is in places where you jump to and there's no friendly planets anywhere. Then You'd die.
Or You could get rescued by some race with jump drives once and they tell you you should not venture there if you aren't sure you'll return. Next time you die.

>> copied my post contents from #3080

**Comment by jafdy** [comment]
Or give escape pods a "miniaturized hyper/jump drive" (based on what the origin ship had perhaps). You could say that creating a drive for a single pod was much easier than creating a drive for a full sized ship.

Or you could give them all jump drives and say that the Quarg build the factories for the systems and blah blah. . . That would raze too many questions though.

**Comment by sivael** [comment]
or that the exploding ship uses the current drive to propel the pods to the neighboring system. Would not solve anything though if you jumped to a system that is not reachable :P
I'd still make it a random amount of time until you get picked up in jumpable spave and give the pods basic hyperdrive capability.

**Comment by tehhowch** [comment]
Pods having hyperdrives makes one question why fighters and drones don't have/use hyperdrives :)

**Comment by sivael** [comment]
True. Might have to wait to be picked up then.

**Comment by XHawk87** [comment]
You could also be rescued by other ships in your fleet

**Comment by Amazinite** [comment]
@tehhowch Well they can, they just aren't build in and usually result in your weapons and engines having to be small in order for them to fit. :)
I imagine that an escape pod would be a swift, weaponless (allowing space for the hyperdrive), fighter-sized ship with the capability to make one or two hyperspace jumps (or a single jump drive jump, if we're able to customize our escape pods), allowing it to at least escape the system and the danger that destroyed its parent ship.

**Comment by jafdy** [comment]
I would argue for it to be a special entity, no one has a good reason to shoot a an escape pod. So, they aren't targeted by anyone or hit by weapons (but can still be hit by blasts), and they interact with friendly ship that will take them back to a friendly port. If you wanted you could have an "executioner" type personality that will shoot escape pods.

**Comment by XHawk87** [comment]
@jafdy I think any situation that would be game ending like that ought to be considered carefully for balance. The danger of any situation in which you might encounter an "executioner" is greatly magnified, but only for players who are taking a hardcore approach to the game. This would make it a high-risk, low reward activity for hardcore players, and they'd avoid it at all costs, unless you also increased the rewards. However, if you did increase the rewards, the normal players would suddenly find the situation to be a low-risk, high-reward activity, and do it to the exclusion of any other. Unless there's a separate hardcore mode which is balanced separately (a lot of extra work), I could see things like that quickly becoming a balance nightmare.

**Comment by sivael** [comment]
I would still vote for an emergency warp drive, one use. That would get the guy out of the area while not being too impossible.
It just burns out after one use. Might even be a shell around your pod that does the "one jump" emergency escape device that pops off glowing red and broken and explodey while your pod uses conventional thrusters to get you somewhere relatively safe to be picked up.

**Comment by jafdy** [comment]
@XHawk87 You seem to be confused between my last sentence and the rest of the paragraphs. Escape pods are a no-reward activity because you just lost your ship, all it will do is let you continue without reloading a save.
I would not encourage use for an "executioner" but in some cases that may be what is wanted for some as-of-yet unknown reason.
Most of my argument is to make an escape pod something that is relatively safe, simply because attacking an escape pod is like shooting a life boat in today's terms.

**Comment by XHawk87** [comment]
@jafdy I'm not confused. I was only talking about the concept of "executioners" and how they would make game balance difficult. If they were only used sparingly as you have now clarified, such as in some unique story-based event, that would be less of an issue, however if they were part of any normal game activity, then it becomes a problem to have ships that try to shoot escape pods. 

As for the rest of your post, I completely agree. Escape pods should generally not be targetted. Although I think escape pod capture instead of killing could lead to some interesting gameplay, such as: turning in pods for bounty, being tried and fined or imprisoned for X days and released, becoming a prisoner of war, being rescued or engineering a prison break. Lots of interesting things could happen with capturing.

**Comment by Thunderforge** [comment]
Escape pods in Escape Velocity were much simpler than what is being described. If your ship was destroyed, a small escape pod went out from the explosion, then you saw a dialog saying something like this: “You remain adrift for a few days before a friendly ship came by and dropped you off at a friendly planet.” I don’t remember how the planets were decided, but I think it was determined by a list of neutral/friendly planets with a shipyard.

That makes things much simpler: no worries about burned out jump drives and players getting stuck, no worries about writing AI to ignore escape pods, etc. Just a hand wave that says you got somewhere safe and can continue on. 

If we are implementing escape pods for hardcore players wanting a roguelike experience, I would advocate for doing the same.

**Comment by XHawk87** [comment]
@Thunderforge That sounds like a great place to start. Over-planning is one of the biggest problems in game design. I think the more complex suggestions should be considered if and when it is called for, but there is no reason not to discuss it and come up with a pool of ideas ready to draw from later in case they are needed.

**Comment by jschiav** [comment]
I dug up my old github account JUST to comment on this issue. I would absolutely play a Hardcore mode. It was the only way I ever played EVN. In fact, I would go as far as to say that a hardcore mode would go a long ways towards bridging a major problem with Endless Sky: that death and failure has almost literally no impact. Since the game's save system is so robust, you can retry a hard mission or even a challenging jump as many times as you want. In fact, it got so immersion-breaking for me, I began disowning my ships when they got destroyed just to make things interesting.

Escape pods are actually the single most valuable game addition I would like to see.

I acknowledge that not every player is like me, but I absolutely guarantee you that many people are, especially in the space combat and trade simulation genre. After all, RISK is one of the key narrative selling points. Without risk, it's just a lot of pew pew, which, while entertaining, eventually gets old for a lot of players.

Also, with regards to whether to kill past saves: my recommendation is to delete all saves that the player hasn't specifically created. If a player creates their own saves, there's probably a specific reason they're doing so (like mission branches as mentioned); if they're abusing it, that's their prerogative, since they're only making things harder for themselves by playing a hardcore mode anyways.

**Comment by Darklocq** [comment]
My two cents:

Already made extensive difficulty-settings commentary at #1386.

On "hardcore mode" ideas:
* Auto-landing off option: YES.  I find myself not actually paying much attention when jumping, because pirate attacks have been infrequent enough that my eyes just kind of glaze over.  Also, if auto-landing is turned off, one's jump point into a system should be farther out, so you can survey the system worlds/stations as you approach, and get a sense of what is going on in the system and what they're doing.

** Please also add an option to speed up (or even eliminate) the jump animation, between the jump actually being initiated and being completed (not the positioning lead-up to it). The EVN mod to do that was one of the most popular, for a reason.  All these jumps actually suck up a lot of human realtime, and they're repetitive; once you've seen a few they're not interesting.  This is maybe worth a feature request on its own, since it's not really tied to auto-landing.

* Escape pods: YES.  One thing I liked about EVN and its predecessors was the more-or-less permanent death.  I kept my saves under version control, just in case I felt I was killed due to a game glitch or something else out of my control, but the financial risk of ship loss was always a big part of that game, as well as the possibility of just being killed despite having an escape pod.  THIS game already has save backups.  It gets tedious and kind of a "god mode" thing to keep just re-loading from my last planetfall any time I blow up.  Robs the game of a lot of tension and such.  If we had an escape pod feature (please don't make it take up much outfit room! ha ha) then many of us would be inclined to treat a legit ship loss as such and not keep reloading.  I'd feel more emotionally invested in the game if doing a bad job at being an adventurer-pilot had painful in-game consequences.

I don't agree that the game needs to auto-delete saves when such a feature turned on.  Maybe have that as an off-by-default option. But it's not the devs' responsibility to code for people who have some kind of self-control disorder and cannot make themselves not cheat.  (Most popular games, going back to DOOM at least, have had some kind of god-mod cheat code, yet most of us didn't run around invincible.  Likewise, I'm a minor modder for Skyrim and have a bunch of dev/testing mods installed that are very easy to cheat with in regular play, but I don't.)  Trust the individual gamer to know how they want to play.  I think a forced auto-delete in hardcore mode would be dangerous, because it would not account for poorly coded mods. E.g., I found that The Station of Dr. Rousseau (also part of Spiral) has a powerful pirate in it that goes after you as soon as you leave your first planet. Some kind of joke on the part of the mod dev.  Took me over an hour of grinding and grinding some basic leave-and-flee tactics to get away from him (only to have him show up at the next planet, too). That would have been something like 50 permanent deaths, so a whole bunch of grinding of new-pilot setup, too.  I.e., please do not box the player in with hard-coded things like clearing old saves.

On implementation: as Thunderforge said above: "Over-planning is one of the biggest problems in game design."  Apply the KISS principle when adding a feature like escape pods.  And this: 'Escape pods in Escape Velocity were much simpler than what is being described [above]. If your ship was destroyed, a small escape pod went out from the explosion, then you saw a dialog saying something like this: “You remain adrift for a few days before a friendly ship came by and dropped you off at a friendly planet.”' Yep. I think EVN at least had the possibility of your escape pod itself being blown up, if you ejected right into the path of a missile or something.  Unless the ES devs make the pod invincible (or not really a game object but an animation and a teleport-to-somewhere script), this would be another reason to not nuke all the old saves.

Other bits, briefly:
* Please do not DISABLE save points (except maybe as an off-by-default option).  There are too many instabilities and variables (especially with mods) for that.  It's fine for this game to be Rogue-lite or Rogue-like-like, not seriously Rogue-like.  In today's world, we have all kinds of additional distractions, like cell phones going off, and pop-up notices injected by our OS or other applications; we cannot always be 100% glued to activity going on in a game.  People should not be punished for having lives.

* I agree with the previous comments about a separate rationale for not deleting old saves, either: many of us keep a "library" of them to explore different choices.

* Another is that if you lose a quest-related passenger or cargo item, that may break the game (or mod). Maybe you have a mod with 52 quest events over a sweeping story arc, and you can't even start it until you're already commanding a capital ship.  You shouldn't have to play for a month with a new pilot just to get back to where you can start that mod again.

* I have no issue with the idea of creating a manual saves system, and maybe blowing away old auto-saves only.  Anyone used to a game like Skyrim is used to doing manual saves pretty often. I'm also reminded of the "rare save points" feature of some games like Borderlands, and going back to Marathon and earlier.  You could go back and recover, but had to re-play significant portions of the game. Having a manual save system would mostly simulate that: If blowed up, you lose progress back to your last manual save; for most of us that would be anywhere from 10 minutes to several hours (people who obsessively manually save every time they turn around are already in their own little self-made Hell, and the devs need not try to account for them).

* I agree with above criticisms about the "executioner" idea and other unnecessary complications. They make it hard to implement, and harder to balance.

* Upon your ship asploding, you should lose it and the cargo, passengers, crew, etc., but not your credits. This game has a banking system for a reason, even if it's under-utilized.

* I didn't read every single word of the above, but one thing I didn't recall seeing (and don't recall from EV mechanics) is what would happen if your flagship blew up but you had a fleet. I would think your closest hyperdrive-capable vessel should pick you up and become your new flagship-for-now, instead of you regenerating on some nearby non-hostile planet with a shipyard. That could perhaps happen in realtime with the fight still ongoing. (And additional risk – did you even remember to put escape pods in your other ships? You might actually get killed if you don't retreat quickly!)  I suppose in that case, it would need to be another scripted transition kind of thing, or an actual in-game-object pod that's not going to blow up. For simplicity, it could just be made non-targetable – too small for anyone to notice or care about, easily mistaken for flotsam, except by your own fleet picking up your encrypted distress call? Or at least non-destroyable, perhaps for X amount of time?

**Comment by Zitchas** [comment]
I'm not really a fan of perma-death (I'll never enabled it). But I'd love to have escape pods in-game.

Ideally speaking, I think the default course of action would be that I'm ejected in an escape pod that one of my escorts tries to pick up (if not hindered by battle), at which point that ship becomes my new flagship and the game proceeds as normal.

If I have no other ships in my fleet that I own, then my escape pod drifts for a short time (30secs max), then fade to black and I'm on the nearest non-hostile planet with a shipyard. Just to prevent problems, there would be somekind of check to ensure that the player can buy at the very least a shuttle (so, something like "is credit total enough to buy the cheapest ship in this shipyard?" If not, do they have access to enough potential loans to buy the cheapest ship? If no, auto-loan them enough to buy the cheapest available ship.

**Comment by Darklocq** [comment]
Yep. Good point about the funding.  Need to be able to at least get another basic shuttle like you start the game with.  How ignominious that would be!  "I lost my 5mil ship and now I'm back in this flying van?"

**Comment by XHawk87** [comment]
@Zitchas 
> If I have no other ships in my fleet that I own, then my escape pod drifts for a short time (30secs max), then fade to black and I'm on the nearest non-hostile planet with a shipyard

I would make it very short in those circumstances, maybe just 3 seconds. Waiting too long could get boring if nothing is happening. However, if there are still ships around that could rescue, I think it'd be better to let the player have agency in deciding when they want to skip ahead to the nearest friendly shipyard, so that it is always the right amount of time for them.

**Comment by Thunderforge** [comment]
Since there are a couple of ideas floating around for what a "hardcore" mode means, why not just make each of them an individual option that players can toggle to their liking.

 - Death is permanent
 - One save point
 - Auto-landing

Maybe one user likes having just one save point (and trusts they won't run into engine bugs), but wants to be able to reload if they die. Or they want an Escape Velocity-style experience with permanent death and no auto-landing, but plan to use multiple save points at certain points according to their challenge preferences.

This seems better to me than a single "hardcore mode" that makes multiple changes.

**Comment by Zitchas** [comment]
Yeah, 30 secs would be too long. I might go for 5 secs if the player has no ships to pick them up. (or until a key is pressed. I could see wanting to keep watching whatever battle killed me for a bit)

A toggle to disable auto landing would be nice.

**Comment by Amazinite** [comment]
On the topic of save deletion: Some people in this issue have said that they don't want saves to be deleted on death, but I would ask, then what's the point of a permadeath mode if it isn't actually permadeath? If you don't want your saves deleted then simply don't play the permadeath mode. But I would say that integral to a permadeath mode is that once you're dead, that's it, and the game doesn't give you any way to go back. (Important note that the *game* shouldn't *give you* a way to go back. That's a separate idea from the game *preventing you* from going back, which I'll discuss further down.)

On how that would actually function, I was playing a game relatively recently that had save files that only updated after a day had passed, similar to how our save files only update on landing, and the way that game worked was that your save files for the permadeath account (in our case pilot) were deleted after you started the day and resaved upon finishing the day. This prevents any sort of save scumming where you reload the save just before actually dying, so I imagine that we could do something similar, or perhaps instead of outright deleting saves just not allowing the player to load a save while in flight within the permadeath mode. Then upon player death is when we actually pull the plug on the saves. The latter idea of just not allowing save loads during flight would allow the player to maintain their previous saves and any snapshots they created during the game, but those would obviously also be deleted upon player death.

The only thing this doesn't defend against is a player making a copy of their save file somewhere else and reimporting it into the game in the scenario where they died, but at that point that player is going above and beyond to get around the permadeath behavior and might as well not even be playing the mode, so I wouldn't worry about trying to "fix" such behavior. Although with #5517, I imagine we could log a number of how many permadeath pilots the player created and how many were deleted (either through player action or through player death) and be able to tell the player, "Hey, we notice that you have a number of permadeath pilots created <= the number of permadeath pilots lost. Very unfortunate," or something similar. Just a sly 4th wall break for people cheating in permadeath mode.

And while I'm here, we have a potential escape pod implementation in #5977 that is currently a WIP. Given that we basically agree that escape pods are necessary for a permadeath mode, I'm interested in getting feedback on that potential implementation from anyone who spoke on this issue.

**Comment by Zitchas** [comment]
While I like the idea of perma-death, I also don't want a system that works extra-hard to make it permanent. As in, I'm not actually a fan of permadeath, but I do like hardmode. For example, having a character that gets backed up at key points so I don't lose *everything* when I die, but there's no "Oh, you can reload from the last place you landed" convenient-resurrection option.

I'd be the sort of person that plays in hardmode, but makes a backup of my character every hour or two. I still have the incentive to do my best to survive, but I have zero interest in being forced to restart from scratch every time I make a mistake.

**Comment by Zitchas** [comment]
Actually, in the ES context, having an option that only saves when the player completes key missions or milestone saves. In timer terms, having something that does a save on the 1st of each month or something like that would also be good.

**Comment by XHawk87** [comment]
I'd agree that going to extreme lengths to delete saves is a mistake, mainly because it'd have side effects like making game crashes and save corruption a bigger problem if they occur. The goal should be to enable players to play the way that they want to. If you're actively fighting to prevent them from doing what they want to do, you're wasting time you could have put into other things. 

I think renaming saves so that they no longer appear in game would probably be good enough for most players, and for those who really can't trust themselves not to try to recover those and actually regret it later, they can always use a script to delete them automatically, if that's what they want.

**Comment by ziproot** [comment]
> Actually, in the ES context, having an option that only saves when the player completes key missions or milestone saves
Hardcore mode could only create autosaves and not any other saves. This could be combined with Amazinite's implementation, where an extra save is created when taking off and deleted when landing or dying, so that players can leave the game in between autosaves.

**Comment by Zitchas** [comment]
To be honest, I rather like having escape pods just for in-universe realism. I'll even equip them on my own flagship and escorts regardless of whether or not there is a mechanical reason (like the risk of permadeath) forcing me to do so.

**Comment by Amazinite** [comment]
A permadeath mode will be implemented with #12345. Escape pod conversation can occur in #1150. Other features discussed here have either been added over the years, or we probably have another open issue for them somewhere else.
